core/amber/src/main/python/core/proxy/proxy_server.py

```diff
@@ -0,0 +1,237 @@
+import argparse
+import threading
+import time
+from functools import wraps
+from inspect import signature
+from typing import Callable, Dict, Iterator, Optional, Tuple
+
+from loguru import logger
+from overrides import overrides
+from pyarrow import Table, py_buffer
+from pyarrow.flight import Action, FlightDescriptor, FlightServerBase, MetadataRecordBatchReader, Result, \
+    ServerCallContext
+from pyarrow.ipc import RecordBatchStreamWriter
+
+
+class ProxyServer(FlightServerBase):
+    """
+    There are three kinds of messages supported by the ProxyServer:
+    1. Data Messages.
+        Data Messages are passed through the endpoint do_put. It will contain a command and a data batch
+        The user should provide deserializer for the command and a handler for the data batch.
+    2. ProxyInternal Messages.
+        ProxyInternal Messages are passed through the endpoint do_action. It will be used to control the
+        life cycle of the ProxyServer. Some example messages:
+        - heartbeat: checks if the ProxyServer is alive.
+        - shutdown: shutdown the ProxyServer.
+        - control: passing a Control Message.
+    3. Control Messages.
+        Control Messages are passed through the endpoint do_action. It will contain a command and a payload.
+        The user should provide deserializer for both the command and the payload.
+    """
+
+    @staticmethod
+    def ack(original_func: Optional[Callable] = None, msg="ack"):
+        """
+        decorator for returning an ack message after the action.
```

Comment:
I was a bit confused here until reading the usages of `ack`, it's easily confused with RPC-level ack. Can you add one line explaining this is proxy-level ack, not rpc-level ack?

Reply:
Also I believe only proxy will use this ack static function / decorator. If I'm correctly, it's good to also add it's intended to only be used within proxy.

Reply:
I added some comments about it. 

Regarding this "ack", I think it might be redundant and can be removed in the future. Java side sends an Action and expects the response to be "ack" indicating a successful call.